components/DeviceGroupModal.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useState } from "react";
+import {
+  Input,
+  Button,
+  Notification,
+  Modal,
+  Form,
+} from "@canonical/react-components";
+
+import { createDeviceGroup } from "@/utils/createDeviceGroup";
+
+function isValidIP(ip: string) {
+  const regex =
+    /^(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)$/;
+  return regex.test(ip);
+}
+
+function isValidMTU(mtu: number) {
+  return mtu >= 1200 && mtu <= 65536;
```

Comment:
This is a 16 bit field, thus the maximum value is 65535.
```suggestion
  return mtu >= 1200 && mtu <= 65535;
```